StandaloneMmPkg/Core/StandaloneMmCore.c

```diff
@@ -559,16 +579,15 @@ MmEntryPoint (
           EFI_PAGES_TO_SIZE (mMmCommunicationBuffer->NumberOfPages) - BufferSize
           );
 
-        CommunicateHeader = (EFI_MM_COMMUNICATE_HEADER *)mInternalCommBufferCopy;
-        BufferSize        = CommunicateHeader->MessageLength;
-        Status            = MmiManage (
-                              &CommunicateHeader->HeaderGuid,
-                              NULL,
-                              CommunicateHeader->Data,
-                              &BufferSize
-                              );
+        BufferSize -= CommHeaderSize;
+        Status      = MmiManage (
+                        CommGuid,
+                        NULL,
+                        CommData,

```

Comment:
Bug here!!! CommData should be pointed to the mInternalCommBufferCopy, which is the shadow MM Communication Buffer, not the original MM Communication Buffer from non-MMRAM.

Reply:
have you passed the standalonemm test? above bug will block the standalonemm solution.

Reply:
Doing re-testing. Will update the PR shortly.

Reply:
Fixed after testing. Thanks for catching it.

Reply:
I also took the chance to make the x64 ovmf platform a standalone mm based platform. Will create a PR separately once I clean up the code.